core/tests/webdriverio_desktop/blogDashboard.js

```diff
@@ -98,6 +98,8 @@ describe('Blog dashboard functionality', function() {
     await blogDashboardPage.navigateToPublishTab();
     await blogDashboardPage.expectNumberOfPublishedBlogPostsToBe(1);
     await blogDashboardPage.unpublishBlogPostWithTitle('Sample Blog Post');
+    await blogDashboardPage.navigateToDraftsTab();
+    await blogDashboardPage.navigateToPublishTab();
```

Comment:
If we are not adding these lines we are getting the below error:
![Screenshot from 2022-07-26 12-30-46](https://user-images.githubusercontent.com/86483059/180943987-cd581708-a867-4b41-9dea-5c0c428f788c.png)

What I concluded was the all selector is still selecting all the blog post that was present previously so I added navigation to update the DOM view and hence all selector now returns the correct value.

Reply:
But why it's not selecting all the blog posts? Is this an issue with the frontend?

Reply:
I don't have idea about the frontend code but everywhere we are using navigation before counting the posts so I also did the same.

Reply:
This might be because of speed of webdriverio, here all selectors is called before DOM gets updated, so we need to add these steps to update the DOM before we call all selectors.

Reply:
[waitForPublishedBlogPostsToLoad()](https://github.com/oppia/oppia/blob/eaeaac61fdc75ad8466c59c3dcb1b4cd07e95e76/core/tests/webdriverio_utils/BlogDashboardPage.js#L228) only waits for the blog post container to show up, it doesn't wait particularly for the amount of blog post that should be present. It means even if the DOM is not updated it will have blogPostContainer present and hence there will be no wait time as its already present.